cpp/src/io/orc/stripe_data.cu

```diff
@@ -1333,7 +1333,7 @@ static __device__ void DecodeRowPositions(orcdec_state_s *s, size_t first_row, i
   }
   while (s->u.rowdec.nz_count < s->top.data.max_vals &&
          s->top.data.cur_row + s->top.data.nrows < s->top.data.end_row) {
-    uint32_t nrows = min(s->top.data.end_row - s->top.data.cur_row,
+    uint32_t nrows = min(s->top.data.end_row - (s->top.data.cur_row + s->top.data.nrows),
```

Comment:
is this change related?

Reply:
This is not exactly related to this issue, I came across this while debugging. nrows left to read should be `s->top.data.end_row - (s->top.data.cur_row + s->top.data.nrows`, since this a while loop and s->top.data.nrows gets update in every loop.

Reply:
is it possible to add a test that fails without this change?

Reply:
Don't think so, it is just to reduce number of iteration which would later get updated again.